docs/how_tos/adding_custom_fields_to_the_registration.rst

```diff
@@ -0,0 +1,138 @@
+
+In order to add fields that already exist in the user model, it is enough to redefine several constants. How to do this will be described in instructions **A**. If you need to add fields that are not in the user model by default, then you need to use instruction **B**.
+
+**A. Add fields that are already exists as column in user profile model**
+-------------------------------------------------------------------------
+You need to redefine several constants in the settings. Several methods can be used for this:
+
+#. **Redefine constants using Django admin and `Site configurations` API. (recommended):**
+    - Go to `Site configurations` in admin: {{LMS}}/admin/site_configuration/siteconfiguration/
+    - Add new settings to OrderedDict (create new `Site configurations` if it wasn't before):
+
+    .. code-block:: json
+
+        {
+            "ENABLE_DYNAMIC_REGISTRATION_FIELDS": "true",
+            "MFE_CONFIG": {
+                "ENABLE_DYNAMIC_REGISTRATION_FIELDS": "true"
+            },
+            "REGISTRATION_EXTRA_FIELDS": {
+                "country": "required",
+                "gender": "optional"
+            }
+        }
+
+    - All possible fields can be find in `EXTRA_FIELDS <https://github.com/openedx/edx-platform/blob/a9355852edede9662762847e0d168663083fc816/openedx/core/djangoapps/user_authn/api/helper.py#L20-L39>`_.
+    - REST API gets cached, if you are in hurry though you can do this command: `tutor local exec redis redis-cli flushall`.
+    - Or you can wait a few minutes until the cache is invalidated.
+    - After this, the new fields should appear on the Auth page.
+    - Required fields will appear on the registration form and optional fields will appear on progressive profiling form.
+
+#. **Also it is possible to redefine settings above with using the Tutor plugin:**
+    - There is tutorial `how Tutor plugin can be created <https://docs.tutor.edly.io/tutorials/plugin.html#creating-a-tutor-plugin>`_.
+    - Settings above should be overriden in the Tutor plugin.
+
+#. **For the local development or testing settings can be overriden in configuration files:**
+    - For example, constants can be added here: `env/apps/openedx/settings/lms/development.py`:
+
+    .. code-block:: python
+
+        ENABLE_DYNAMIC_REGISTRATION_FIELDS = "true"
+
+        MFE_CONFIG["ENABLE_DYNAMIC_REGISTRATION_FIELDS"] = "true"
+
+        REGISTRATION_EXTRA_FIELDS["country"] = "required"
+
+        REGISTRATION_EXTRA_FIELDS["gender"] = "required"
+
+`In total, you must redefine 3 constants using the method that is most preferable for you:`
+
+    .. code-block:: python
+
+        ENABLE_DYNAMIC_REGISTRATION_FIELD = True,
+
+        MFE_CONFIG["ENABLE_DYNAMIC_REGISTRATION_FIELDS"] = True,
+
+        REGISTRATION_EXTRA_FIELDS["field_name"] = "required/optionl/hidden"
+
+
+**B. Add fields that do not exist in the user profile model**
```

Comment:
Hello @GlugovGrGlib 

I was working on an extended registration form for an openedx instance and face some issues to make it work along with the authn MFE. While doing some research on the possible workarounds I came across with this PR.

Just want to know if you have had the opportunity to test the implementation of an extended registration form along with this MFE? Because I found a conflict between the django model/form definitions and the MFE structure and the way the MFE handles the fields definition. 

The purpose of this whole comment is to double check if the steps documented in this B section were already proven since I've follow the exact same steps and haven't make it work, so it would be cool if we confirm that the Authn MFE supports the extended forms or if it's a future work needed to be doing in another PR.